### PR TITLE
Home page

### LINKED ISSUE #7: Home page
Image

### PR DESCRIPTION
**Pull request by pantoja**
## What I did
Creates home page with fixed content.
image
image

## Missing content:
- Yet to add quests logic
- Add points badge
- Test dynamic content

Closes #7

### CODE REVIEW

File: src/components/ProgressBar/index.tsx [hunk 1]
```diff
@@ -4,10 +4,16 @@ import ProgressBar from './ProgressBar';
 
 interface WrapperTypes {
   progress: number;
+  progressTitle: string;
 }
 
-const Wrapper: FC<WrapperTypes> = ({ progress }) => {
-  return <ProgressBar progress={progress} />;
+const Wrapper: FC<WrapperTypes> = ({ progress, progressTitle }) => {
```
Comment (KevBeltrao) [inline src/components/ProgressBar/index.tsx:10]: The wrapper (index.tsx) should container interface content (`div`, `p`, etc). Its goal is actually to add logic and hooks that are gonna be passed as props to the interface.

Check out the QuestCard component as example:

`index.tsx`

image

It's where we use the `useNavigate` hook to create the `navigate` variable. Is also where is the logic to decide the Icon we should use and also to declare the `handleClick` function. So is where we use all the hooks, declare variables, functions, etc.

This way, QuestCard.tsx holds no logic, just receives everything as props:

`QuestCard.tsx`

image

As a little extra, I'd suggest not adding the <p> to the ProgressBar component, instead leaving just the progress bar itself (so the text would be added on the component importing ProgressBar). If you really want to have the text inside the ProgressBar component since we might not need the isolated bar, I'd still ask you to move the <p> to inside the component.

So in other words:

- Preferable: text outside the ProgressBar component, ProgressBar only exporting the progress bar.

- Acceptable: moving the <p> from the `index.tsx` to the `ProgressBar.tsx`.

File: src/components/ProgressBar/index.tsx [hunk 2]
```diff
@@ -4,10 +4,16 @@ import ProgressBar from './ProgressBar';
 
 interface WrapperTypes {
   progress: number;
+  progressTitle: string;
```
Comment (KevBeltrao) [inline src/components/ProgressBar/index.tsx:7]: Because of this extra prop, we're getting this complaint on the `src/components/ProgressBar/ProgressBar.test.tsx`

But that should go away if you choose not to use this prop anymore, like I commented it'd be preferable on the `index.tsx`.

image

File: src/pages/Home/Home.tsx [hunk 3]
```diff
@@ -1,15 +1,40 @@
 import { type FC } from 'react';
 
-import { Container } from './styles';
+import Progress from '../../components/ProgressBar';
```
Comment (KevBeltrao) [inline src/pages/Home/Home.tsx:3]: I'd encourage naming the importing variable just like the exported name (ProgressBar instead of Progress)

File: src/pages/Home/Home.tsx [hunk 4]
```diff
@@ -1,15 +1,40 @@
...
 interface HomeTypes {
   initial?: number;
 }
 
 const Home: FC<HomeTypes> = () => {
   return (
     <Container>
-      <h1>Home</h1>
+      <h1>É hora do jogo!</h1>
+      <h2>Complete as quests corretamente e obtenha uma surpresa no final</h2>
```
Comment (KevBeltrao) [inline src/pages/Home/Home.tsx:15]: This h2 should have font-weight 500

File: src/pages/Home/Home.tsx [hunk 5]
```diff
@@ -1,15 +1,40 @@
...
+
+      <ProgressContainer>
+        <Progress progress={0} progressTitle="Quests finalizadas" />
+
+        {/* TODO: Points badge here */}
+        <div> </div>
+      </ProgressContainer>
+
+      <QuestsWrapper>
+        <CardQuest
```
Comment (KevBeltrao) [inline src/pages/Home/Home.tsx:25]: Since you're working on the home page, I'd use the global context instead of mocking the values.

You can see an example of the syntax to consume the global storage on the code I removed on this commit's `src/App.tsx`:
https://github.com/KevBeltrao/conectados/commit/6dedb33ae72057b6065e56f4241473c8cb557539

PS: Add the logic to the `src/pages/Home/index.tsx` and pass the desired values as props.

(Remember to also pass the progress instead of letting `0`).

**Review by KevBeltrao — APPROVED** [review]
